lib/graphql/query.rb

```diff
@@ -246,10 +246,13 @@ def irep_selection
     # @param definition [GraphQL::Schema::Field]
     # @param parent_object [GraphQL::Schema::Object]
     # @return Hash{Symbol => Object}
-    def arguments_for(ast_node, definition, parent_object: nil)
+    def arguments_for(ast_node, definition, parent_object: nil, detailed: false)
       if interpreter?
         @arguments_cache ||= Execution::Interpreter::ArgumentsCache.new(self)
-        @arguments_cache.fetch(ast_node, definition, parent_object)
+        combined_args = @arguments_cache.fetch(ast_node, definition, parent_object)
+        schema.after_lazy(combined_args) do |args|
```

Comment:
Is this here to let you get args after they are mixed in from variables?

Reply:
It's from #2834 -- input_object `loads:` hooks can return promises, which means `coerce_arguments` can return promises, which means `arguments_for` can return promises ...

Reply:
Thanks for explaining!